connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -386,14 +386,46 @@ private[sql] class AvroDeserializer(
       case (LONG, _: DayTimeIntervalType) => (updater, ordinal, value) =>
         updater.setLong(ordinal, value.asInstanceOf[Long])
 
-      case (LONG, _: DecimalType) => (updater, ordinal, value) =>
-        val d = avroType.getLogicalType.asInstanceOf[CustomDecimal]
-        updater.setDecimal(ordinal, Decimal(value.asInstanceOf[Long], d.precision, d.scale))
+      case (INT, dt: DecimalType) => avroType.getLogicalType match {
+        case null =>
```

Comment:
In addition to the INT support, seems we also added a few more checks. Can you say more about it in the PR description?

Reply:
In addition to supporting the use of widened DecimalType to read int and long; I also refer to the previously customized decimal logic type #41409 to read long, which is also supported when reading int.

If you think that these two parts are not suitable to be placed in one PR, I can seperate them apart.

Reply:
Does it introduce any breaking change to end users?

Reply:
All of them are new feature supported.

1. For `int` field in Avro if we use `DecimalType` to read it:
- before, it throws a Exception;
![image](https://github.com/user-attachments/assets/028dc65e-162d-4e93-b876-fd54a5556ec9)

- after, it can works;

And we also supported users to use `DecimalType` with `CustomDecimal` logic Type to read `int` fields.

2. For long field in Avro if we use DecimalType to read it:
- before, it throws a Exception with cause NPE;
![image](https://github.com/user-attachments/assets/8128b887-110f-4300-8d25-ce733d642ef6)

- after, it can works;

And if users use `DecimalType` with `CustomDecimal` logic Type to read `long` fields, the exist feature still works.